src/utils/query.js

```diff
@@ -112,36 +116,89 @@ module.exports = {
 
     const sortAsc =
       (!params.sortAscending && params.previous) || (params.sortAscending && !params.previous);
-    const comparisonOp = sortAsc ? '$gt' : '$lt';
 
     // a `next` cursor will have precedence over a `previous` cursor.
     const op = params.next || params.previous;
 
     if (params.paginatedField == '_id') {
-      return {
-        _id: {
-          [comparisonOp]: op,
-        },
-      };
+      if (sortAsc) {
+        return { _id: { $gt: op } };
+      } else {
+        return { _id: { $lt: op } };
+      }
     } else {
       const field = params.sortCaseInsensitive ? '__lc' : params.paginatedField;
-      return {
-        $or: [
-          {
-            [field]: {
-              [comparisonOp]: op[0],
-            },
-          },
-          {
-            [field]: {
-              $eq: op[0],
-            },
-            _id: {
-              [comparisonOp]: op[1],
-            },
-          },
-        ],
-      };
+
+      const notUndefined = { [field]: { $exists: true } };
+      const onlyUndefs = { [field]: { $exists: false } };
+      const notNullNorUndefined = { [field]: { $ne: null } };
+      const nullOrUndefined = { [field]: null };
+      const onlyNulls = { $and: [{ [field]: { $exists: true } }, { [field]: null }] };
+
+      switch (op[0]) {
```

Comment:
Can we name those `op[0]` and `op[1]`?

Reply:
Good idea. I've named them `paginatedFieldValue` and `idValue`.